### ISSUE TITLE
Helm install fails for charts with Jobs, E1009 field is immutable

### ISSUE THREAD

**Issue by dacleyra** [istio/issues#93]
**Is this a BUG or FEATURE REQUEST?:** Bug

**Did you review existing epics or issues to identify if this already being worked on?** Yes

**Bug:** Y

**What Version of Istio and Kubernetes are you using, where did you get Istio from, Installation details**
```
istioctl version
Version: 0.2.7
GitRevision: 6b145c189aad8306b13af1725123bebfbc7eefd4
GitBranch: master
User: root@f1eeb85f62ab
GolangVersion: go1.8.3

kubectl version
[root@dacleyradev icp]# kubectl version
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.0", GitCommit:"6e937839ac04a38cac63e6a7a306c5d035fe7b0a", GitTreeState:"clean", BuildDate:"2017-09-28T22:57:57Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.3-7+154699da4767fd", GitCommit:"154699da4767fd4225cbaa91cc26abd71bc853c7", GitTreeState:"clean", BuildDate:"2017-08-28T06:41:56Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

```

**Is Istio Auth enabled or not ?** No
`curl -L https://raw.githubusercontent.com/istio/istio/master/release/downloadIstioCandidate.sh | sh -`
`cp -f istio-*/bin/istioctl /usr/local/bin/istioctl`
`kubectl apply -f istio-*/install/kubernetes/istio.yaml`
`kubectl apply -f istio-*/install/kubernetes/istio-initializer.yaml`

**What happened**:
Attempting to install a helm chart that contains a Job fails.
The initializer pod log contains an E1009 with field is immutable message
[initializer.txt](https://github.com/istio/issues/files/1369328/initializer.txt)

**What you expected to happen:**
The helm install of the chart succeeds

**How to reproduce it:**
Install Istio
Install the microservice builder fabric helm chart, which contains a job
helm repo add mb http://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/microservicebuilder/helm/
helm install mb/fabric

**Feature Request:**
N

**Describe the feature:**

**Comment by linsun** [comment]
Thanks for reporting this. I'll try recreate it today.

**Comment by gyliu513** [comment]
As a workaround first, you can disable the initializer to update the `Job`, you can run `kubectl edit initializerconfigurations` and delete the objects in red line.

![image](https://user-images.githubusercontent.com/4461983/31362461-3fe0f79c-ad1e-11e7-864a-37fec647d55f.png)

**Comment by linsun** [comment]
I'm getting this error during recreate:

```Lins-MacBook-Pro:Downloads linsun$ helm install mb/fabric
Error: release doltish-nightingale failed: Timeout: request did not complete within allowed duration```

I did notice the pods/service/deployment are created fine but no jobs.

I agree that the `kind: job` is causing the issue.  Why would we even want to enable job for auto side car injection by default?   could you even expose these as k8s service?  maybe @ayj can shed some light.

**Comment by davidcurrie** [comment]
I can imagine that a job might want a proxy for outbound traffic.

**Comment by ldemailly** [comment]
please use the instructions on https://istio.io/docs for now

**Comment by linsun** [comment]
Ok good point @davidcurrie, it is not useful for inbound but could be useful for outbound.

**Comment by raudaviciusm** [comment]
@dacleyra @Alexander-Owen-Meehan

I noticed that you are using a different fabric repository to the one we are maintaining. Could you please try to run the whole thing by using https://github.ibm.com/liber8/fabric-kubernetes and confirm that the issue still exists.

**Comment by dacleyra** [comment]
The internal master fabric is what I test with normally, and exhibits the same problem with the Job.
I thought it prudent to demonstrate the issue with the published 1.0.0 for the purpose of this issue.

**Comment by raudaviciusm** [comment]
@dacleyra We could not recreate the problem exactly you specified above, however, I have added the annotation to avoid sidecar injection into the job object. The fix is in https://github.ibm.com/liber8/fabric-kubernetes/tree/fix/issue_103 . We believe it should fix the issue. Could you please run the whole thing again by installing fabric from the branch above?

**Comment by ayj** [comment]
This issue is not specific to helm. I've created a new issue in istio/pilot that is specific to initializer's handling of jobs (see https://github.com/istio/pilot/issues/1439).

**Comment by GregHanson** [comment]
helm integration is implemented as of istio 0.5.1 or later.  further work can be tracked at this issue: https://github.com/istio/istio/issues/2516
